dbms/src/Storages/DeltaMerge/BitmapFilter/BitmapFilter.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/DeltaMerge/BitmapFilter/BitmapFilter.h>
+#include <Storages/DeltaMerge/DeltaMergeHelpers.h>
+#include <Storages/DeltaMerge/Segment.h>
+
+namespace DB::DM
+{
+BitmapFilter::BitmapFilter(UInt32 size_, const SegmentSnapshotPtr & snapshot_)
+    : filter(size_, false)
+    , snap(snapshot_)
+    , all_match(false)
+{}
+
+void BitmapFilter::set(const ColumnPtr & col, const FilterPtr f)
+{
+    const auto * v = toColumnVectorDataPtr<UInt32>(col);
+    set(v->data(), v->size(), f);
+}
+
+void BitmapFilter::set(const UInt32 * data, UInt32 size, const FilterPtr f)
+{
+    if (size == 0)
+    {
+        return;
+    }
+    //size_t max_row_id = *std::max_element(data, data + size);
+    //RUNTIME_CHECK(max_row_id < filter.size(), max_row_id, filter.size());
+    if (f == nullptr)
```

Comment:
`f == nullptr` means all positive seems not human-friendly to me.
Besides, better use `!f` instead of `f == nullptr`.

Reply:
You can use another function like `setAll` to make all rows positive. And do this logic outside of this class.